doc/reference/configuration/configuration_reference.rst

```diff
@@ -2889,6 +3004,40 @@ This section describes configuration parameters related to roles.
 
     **Since:** :doc:`3.0.0 </release/3.0.0>`.
 
+    Specify a role's configuration.
+    This option accepts a role name as the key and a role's configuration as the value.
+    To specify the roles of an instance, use the :ref:`roles <configuration_reference_roles>` option.
```

Comment:
We can also say that `validate()` and `apply()` functions accept **this** value[^1] as an argument.

We can also say that this argument is for convenience, but any value from a currently applying configuration may be received from `validate()` and `apply()` using the  `config:get(<...>)` call.

We can also say that a user is not obligated to use only `roles_cfg.<role name>` to store role's configuration. It is not a mistake if `roles_cfg` has a key that doesn't correspond to any existing role name. A role may read anything from `roles_cfg` or from the configuration at all.

However, `roles_cfg.<role name>` is good default choice.

[^1]: To be precise: `roles_cfg.<role name>`.